qutebrowser/browser/urlmarks.py

```diff
@@ -218,6 +215,24 @@ def quickmark_del(self, name):
         except KeyError:
             raise cmdexc.CommandError("Quickmark '{}' not found!".format(name))
 
+    def get_by_qurl(self, url):
+        """Look up a quickmark by QUrl, returning its name.
+
+        Takes O(n) time, where n is the number of quickmarks.
+        Use a name instead where possible.
+        """
+        if not url.isValid():
+            raise ValueError("Invalid URL: {}".format(url.errorString()))
```

Comment:
You can use `qtutils.ensure_valid(url)` here (which will give you a `qtutils.QtValueError`)